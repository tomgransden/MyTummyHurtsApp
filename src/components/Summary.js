import {
  StyleSheet,
  SafeAreaView,
  ScrollView,
  View,
  Text,
  StatusBar,
  Image,
} from "react-native";
import PageHeader from "./PageHeader";
import dayjs from "dayjs";

const types = {
  0: "Medication",
  1: "Food",
};

const exampleData = [
  {
    createdDate: new Date(2023, 1, 2, 3, 24, 0),
    type: 0,
    metadata: {
      medications: ["Buscopan 20mg", "Ibuprofen 40mg", "Paracetamol 100mg"],
    },
  },
  {
    createdDate: new Date(2023, 1, 2, 14, 24, 0),
    type: 1,
    metadata: {
      image:
        "https://img.kidspot.com.au/ykOJKEb7/kk/2019/08/greeneggs-and-ham-601642-1.jpg",
      description:
        "Green eggs and hamasnhasFKSAhab   adsBFGBas asuigF uiasfJBsf gasFBJsa asuIFJAS asuFUasf iuasbwfJF UIafsaf",
    },
  },
];

const renderItem = (item, index) => {
  switch (item.type) {
    case 0:
      return (
        <View
          key={`${index}-${item.type}`}
          style={{
            minHeight: 56,
            paddingBottom: 16,
            marginHorizontal: 16,
            borderWidth: 1,
            backgroundColor: "white",
          }}
        >
          <View
            style={{
              flexDirection: "row",
              marginTop: 8,
              paddingHorizontal: 16,
            }}
          >
            <View
              style={{
                backgroundColor: "white",
                alignSelf: "flex-start",
              }}
            >
              <Text style={{ fontSize: 16 }}>
                {dayjs(item.createdDate).format("hh:mma")}
              </Text>
            </View>
            <Text> - </Text>
            <View
              style={{
                backgroundColor: "white",
                alignSelf: "flex-start",
              }}
            >
              <Text style={{ fontSize: 16 }}>{types?.[item?.type]}</Text>
            </View>
          </View>
          <View
            style={{
              flexDirection: "row",
              flexWrap: "wrap",
              paddingHorizontal: 12,
            }}
          >
            {item.metadata.medications.map((medication, index) => (
              <View
                key={`medication-${index}`}
                style={{
                  borderWidth: 1,
                  borderRadius: 32,
                  marginRight: 12,
                  marginTop: 12,
                  minHeight: 24,
                  padding: 8,
                }}
              >
                <Text style={{ fontSize: 12 }}>{medication}</Text>
              </View>
            ))}
          </View>
        </View>
      );
    case 1:
      return (
        <View
          key={`${index}-${item.type}`}
          style={{
            minHeight: 56,
            backgroundColor: "red",
            marginHorizontal: 16,
            paddingBottom: 16,
            marginTop: 16,
            borderWidth: 1,
            backgroundColor: "white",
          }}
        >
          <View
            style={{
              flexDirection: "row",
              marginTop: 8,
              paddingHorizontal: 16,
            }}
          >
            <View
              style={{
                backgroundColor: "white",
                alignSelf: "flex-start",
              }}
            >
              <Text style={{ fontSize: 16 }}>
                {dayjs(item.createdDate).format("hh:mma")}
              </Text>
            </View>
            <Text> - </Text>
            <View
              style={{
                backgroundColor: "white",
                alignSelf: "flex-start",
              }}
            >
              <Text style={{ fontSize: 16 }}>{types?.[item?.type]}</Text>
            </View>
          </View>
          <View
            style={{
              marginHorizontal: 16,
              flexDirection: "row",
              marginTop: 16,
            }}
          >
            <Image
              source={{ uri: item?.metadata?.image }}
              style={{ width: 60, height: 60, resizeMode: "cover" }}
            />
            <Text style={{ marginHorizontal: 16 }}>
              {item?.metadata?.description}
            </Text>
          </View>
        </View>
      );
  }
};

export default function Summary() {
  return (
    <SafeAreaView style={styles.container}>
      <PageHeader title={"My tummy hurts"} />
      <ScrollView contentContainerStyle={{ marginTop: 16 }}>
        {exampleData.map((item, index) => renderItem(item, index))}
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#bfa2c8",
    marginTop: StatusBar.currentHeight,
  },
});
