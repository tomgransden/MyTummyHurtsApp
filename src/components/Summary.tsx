import dayjs from 'dayjs';
import { StyleSheet, SafeAreaView, ScrollView, View, Text, Image } from 'react-native';

enum Type {
  Medication,
  Food,
}

type MedicationDataPoint = DataPointBase & {
  type: 0;
  metadata: {
    medications: string[];
  };
};

type FoodDataPoint = DataPointBase & {
  type: 1;
  metadata: {
    image: string;
    description: string;
  };
};

type DataPointBase = {
  createdDate: Date;
};

const exampleData: (MedicationDataPoint | FoodDataPoint)[] = [
  {
    createdDate: new Date(2023, 1, 2, 3, 24, 0),
    type: 0,
    metadata: {
      medications: ['Buscopan 20mg', 'Ibuprofen 40mg', 'Paracetamol 100mg'],
    },
  },
  {
    createdDate: new Date(2023, 1, 2, 14, 24, 0),
    type: 1,
    metadata: {
      image: 'https://img.kidspot.com.au/ykOJKEb7/kk/2019/08/greeneggs-and-ham-601642-1.jpg',
      description:
        'Green eggs and hamasnhasFKSAhab   adsBFGBas asuigF uiasfJBsf gasFBJsa asuIFJAS asuFUasf iuasbwfJF UIafsaf',
    },
  },
];

const renderItem = (item: MedicationDataPoint | FoodDataPoint, index: number) => {
  switch (item.type) {
    case 0:
      return (
        <View
          key={`${index}-${item.type}`}
          style={{
            minHeight: 56,
            paddingBottom: 16,
            marginHorizontal: 16,
            borderWidth: 1,
            backgroundColor: 'white',
          }}>
          <View
            style={{
              flexDirection: 'row',
              marginTop: 8,
              paddingHorizontal: 16,
            }}>
            <View
              style={{
                backgroundColor: 'white',
                alignSelf: 'flex-start',
              }}>
              <Text style={{ fontSize: 16 }}>{dayjs(item.createdDate).format('hh:mma')}</Text>
            </View>
            <Text> - </Text>
            <View
              style={{
                backgroundColor: 'white',
                alignSelf: 'flex-start',
              }}>
              <Text style={{ fontSize: 16 }}>{Type[item.type]}</Text>
            </View>
          </View>
          <View
            style={{
              flexDirection: 'row',
              flexWrap: 'wrap',
              paddingHorizontal: 12,
            }}>
            {item.metadata.medications.map((medication, index) => (
              <View
                key={`medication-${index}`}
                style={{
                  borderWidth: 1,
                  borderRadius: 32,
                  marginRight: 12,
                  marginTop: 12,
                  minHeight: 24,
                  padding: 8,
                }}>
                <Text style={{ fontSize: 12 }}>{medication}</Text>
              </View>
            ))}
          </View>
        </View>
      );
    case 1:
      return (
        <View
          key={`${index}-${item.type}`}
          style={{
            minHeight: 56,
            marginHorizontal: 16,
            paddingBottom: 16,
            marginTop: 16,
            borderWidth: 1,
            backgroundColor: 'white',
          }}>
          <View
            style={{
              flexDirection: 'row',
              marginTop: 8,
              paddingHorizontal: 16,
            }}>
            <View
              style={{
                backgroundColor: 'white',
                alignSelf: 'flex-start',
              }}>
              <Text style={{ fontSize: 16 }}>{dayjs(item.createdDate).format('hh:mma')}</Text>
            </View>
            <Text> - </Text>
            <View
              style={{
                backgroundColor: 'white',
                alignSelf: 'flex-start',
              }}>
              <Text style={{ fontSize: 16 }}>{Type[item.type]}</Text>
            </View>
          </View>
          <View
            style={{
              marginHorizontal: 16,
              flexDirection: 'row',
              marginTop: 16,
            }}>
            <Image
              source={{ uri: item?.metadata?.image }}
              style={{ width: 60, height: 60, resizeMode: 'cover' }}
            />
            <Text style={{ marginHorizontal: 16 }}>{item.metadata.description}</Text>
          </View>
        </View>
      );
  }
};

const Summary = (): JSX.Element => (
  <SafeAreaView style={styles.container}>
    <ScrollView contentContainerStyle={{ marginTop: 16 }}>
      {exampleData.map((item, index) => renderItem(item, index))}
    </ScrollView>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#bfa2c8',
  },
});

export default Summary;
