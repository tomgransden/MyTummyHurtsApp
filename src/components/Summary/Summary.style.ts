import { StyleSheet, ViewStyle } from 'react-native';

type SummaryStyle = {
  container: ViewStyle;
  scrollViewContainer: ViewStyle;
};

const styles = StyleSheet.create<SummaryStyle>({
  container: {
    flex: 1,
    backgroundColor: '#bfa2c8',
  },
  scrollViewContainer: { marginTop: 16 },
});

export default styles;
