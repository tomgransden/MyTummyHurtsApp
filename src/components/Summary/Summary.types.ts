import { ViewStyle } from 'react-native';

export enum RecordType {
  Medication,
  Food,
}

type DataPointBase = {
  createdDate: Date;
};

export type MedicationDataPoint = DataPointBase & {
  type: RecordType.Medication;
  metadata: {
    medications: string[];
  };
};

export type FoodDataPoint = DataPointBase & {
  type: RecordType.Food;
  metadata: {
    image: string;
    description: string;
  };
};

export type DataPoint = MedicationDataPoint | FoodDataPoint;

export type SummaryStyle = {
  container: ViewStyle;
  scrollViewContainer: ViewStyle;
};
