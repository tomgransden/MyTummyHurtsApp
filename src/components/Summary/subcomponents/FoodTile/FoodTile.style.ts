import { ImageStyle, StyleSheet, TextStyle, ViewStyle } from 'react-native';

type FoodTileStyle = {
  container: ViewStyle;
  titleContainer: ViewStyle;
  timeContainer: ViewStyle;
  titleText: TextStyle;
  contentContainer: ViewStyle;
  image: ImageStyle;
  description: TextStyle;
};

const styles = StyleSheet.create<FoodTileStyle>({
  container: {
    minHeight: 56,
    marginHorizontal: 16,
    paddingBottom: 16,
    marginTop: 16,
    borderWidth: 1,
    backgroundColor: 'white',
  },
  titleContainer: {
    flexDirection: 'row',
    marginTop: 8,
    paddingHorizontal: 16,
  },
  timeContainer: {
    backgroundColor: 'white',
    alignSelf: 'flex-start',
  },
  titleText: { fontSize: 16, fontFamily: 'Rubik' },
  contentContainer: {
    marginHorizontal: 16,
    flexDirection: 'row',
    marginTop: 16,
  },
  image: { width: 60, height: 60, resizeMode: 'cover' },
  description: { marginLeft: 16, fontFamily: 'Rubik', flex: 1, flexWrap: 'wrap' },
});

export default styles;
