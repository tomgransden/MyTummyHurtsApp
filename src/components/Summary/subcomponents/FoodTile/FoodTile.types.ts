import { ImageStyle, TextStyle, ViewStyle } from 'react-native';

import { FoodDataPoint } from '../../Summary.types';

export type FoodTileStyle = {
  container: ViewStyle;
  titleContainer: ViewStyle;
  timeContainer: ViewStyle;
  titleText: TextStyle;
  contentContainer: ViewStyle;
  image: ImageStyle;
  description: TextStyle;
};

export type FoodTileProps = {
  item: FoodDataPoint;
};
