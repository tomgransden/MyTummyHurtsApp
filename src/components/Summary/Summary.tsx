import { SafeAreaView, ScrollView } from 'react-native';

import styles from './Summary.style';
import { IRecordType, IDataPoint } from './Summary.types';
import FoodTile from './subcomponents/FoodTile/FoodTile';
import MedicationTile from './subcomponents/MedicationTile/MedicationTile';

const exampleData: IDataPoint[] = [
  {
    createdDate: new Date(2023, 1, 2, 3, 24, 0),
    type: IRecordType.Medication,
    metadata: {
      medications: ['Buscopan 20mg', 'Ibuprofen 40mg', 'Paracetamol 100mg'],
    },
  },
  {
    createdDate: new Date(2023, 1, 2, 14, 24, 0),
    type: IRecordType.Food,
    metadata: {
      image: 'https://img.kidspot.com.au/ykOJKEb7/kk/2019/08/greeneggs-and-ham-601642-1.jpg',
      description: 'Green eggs and some tasty ham from my fridge',
    },
  },
];

const renderItem = (item: IDataPoint, index: number) => {
  switch (item.type) {
    case IRecordType.Medication:
      return <MedicationTile item={item} key={index} />;
    case IRecordType.Food:
      return <FoodTile item={item} key={index} />;
    default:
      return null;
  }
};

const Summary = () => (
  <SafeAreaView style={styles.container}>
    <ScrollView contentContainerStyle={styles.scrollViewContainer}>
      {exampleData.map((item, index) => renderItem(item, index))}
    </ScrollView>
  </SafeAreaView>
);

export default Summary;
