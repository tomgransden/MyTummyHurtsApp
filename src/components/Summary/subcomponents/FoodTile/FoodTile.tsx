import dayjs from 'dayjs';
import { View, Text, Image } from 'react-native';

import { RecordType } from '../../Summary.types';
import styles from './FoodTile.style';
import { FoodTileProps } from './FoodTile.types';

const FoodTile = ({ item }: FoodTileProps) => (
  <View style={styles.container}>
    <View style={styles.titleContainer}>
      <View style={styles.timeContainer}>
        <Text style={styles.titleText}>{dayjs(item.createdDate).format('hh:mma')}</Text>
      </View>
      <Text> - </Text>
      <View style={styles.timeContainer}>
        <Text style={styles.titleText}>{RecordType[item.type]}</Text>
      </View>
    </View>
    <View style={styles.contentContainer}>
      <Image source={{ uri: item?.metadata?.image }} style={styles.image} />
      <Text style={styles.description}>{item.metadata.description}</Text>
    </View>
  </View>
);

export default FoodTile;
