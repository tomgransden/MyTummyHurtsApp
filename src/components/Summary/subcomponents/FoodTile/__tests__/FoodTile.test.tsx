import { render, screen } from '@testing-library/react-native';
import React from 'react';

import { RecordType } from '../../../Summary.types';
import FoodTile from '../FoodTile';

describe('FoodTile', () => {
  it('renders correctly', () => {
    render(
      <FoodTile
        item={{
          createdDate: new Date('2023-01-01'),
          type: RecordType.Food,
          metadata: { image: 'string', description: 'Food' },
        }}
      />
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
