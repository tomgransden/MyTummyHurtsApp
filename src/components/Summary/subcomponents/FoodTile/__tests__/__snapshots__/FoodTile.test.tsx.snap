// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodTile renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderWidth": 1,
      "marginHorizontal": 16,
      "marginTop": 16,
      "minHeight": 56,
      "paddingBottom": 16,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 8,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      style={
        {
          "alignSelf": "flex-start",
          "backgroundColor": "white",
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Rubik",
            "fontSize": 16,
          }
        }
      >
        12:00am
      </Text>
    </View>
    <Text>
       - 
    </Text>
    <View
      style={
        {
          "alignSelf": "flex-start",
          "backgroundColor": "white",
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Rubik",
            "fontSize": 16,
          }
        }
      >
        Food
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginHorizontal": 16,
        "marginTop": 16,
      }
    }
  >
    <Image
      source={
        {
          "uri": "string",
        }
      }
      style={
        {
          "height": 60,
          "resizeMode": "cover",
          "width": 60,
        }
      }
    />
    <Text
      style={
        {
          "flex": 1,
          "flexWrap": "wrap",
          "fontFamily": "Rubik",
          "marginLeft": 16,
        }
      }
    >
      Food
    </Text>
  </View>
</View>
`;
